test/beginend-dired-test.el

```diff
@@ -33,80 +33,87 @@
 (require 'dired)
 (require 'dired-x)
 
-(defun beginend-create-dired-buffer ()
-  "Create and return a buffer ressembling a dired one."
-  (assess-with-filesystem '("dir1/" "dir2/")
-    (dired default-directory)
-    (current-buffer)))
-
-(defun beginend-dired-test (&rest spec)
-  "Test dired from SPEC.
-SPEC is a list of symbols representing features to activate."
-  (let* ((begin-p (seq-contains spec 'begin-p))
-         (omit (seq-contains spec 'omit))
-         (hide-details (seq-contains spec 'hide-details))
-         (hide-info-lines (not (seq-contains spec 'keep-info-lines)))
-         (dired-hide-details-hide-information-lines hide-info-lines)
-         (dired-omit-verbose nil))
-    (with-current-buffer (beginend-create-dired-buffer)
-      (goto-char 2) ;; any position
-      (when omit
-        (dired-omit-mode))
-      (when hide-details
-        (dired-hide-details-mode))
-      (if begin-p
-          (progn
-            (beginend-dired-mode-goto-beginning)
-            (expect (looking-at "dir1") :to-be-truthy))
-        (beginend-dired-mode-goto-end)
-        (expect (looking-at "dir2") :to-be-truthy))
-      (kill-buffer))))
+(require 'seq)
+
+(defun beginend-create-dired-buffer (entries)
+  "Create and return a buffer resembling a dired one.
+
+ENTRIES is the file list.  The order is important.  ENTRIES are listed in that
+order."
+  (let ((spec (seq-filter (lambda (x) (not (member x '("." "..")))) entries)))
+    (assess-with-filesystem spec
+      (dired (cons default-directory entries))
+      (current-buffer))))
+
+(defun beginend-dired-test (fn filename)
+  "Execute FN at `point-min' and check that point is before FILENAME.
+This function moves point to `point-min' and executes FN with no parameter.
+Then, if FILENAME is non-nil, checks that point is at the beginning of
+FILENAME.  If FILENAME is nil, checks that point is at `point-max'"
+  (goto-char (point-min))
+  (funcall fn)
+  (if filename
+      (expect (looking-at filename) :to-be-truthy)
+    (expect (eobp) :to-be-truthy)))
+
+(defun beginend-dired-test-begin (filename)
+  "Move the point to the first line of file list.
+Then, if FILENAME is non-nil, checks that point is at the beginning of
+FILENAME.  If FILENAME is nil, checks that point is at `point-max'."
+  (beginend-dired-test 'beginend-dired-mode-goto-beginning filename))
+
+(defun beginend-dired-test-end (filename)
+  "Move the point to the last line of file list.
+Then, if FILENAME is non-nil, checks that point is at the beginning of
+FILENAME.  If FILENAME is nil, checks that point is at `point-max'."
+  (beginend-dired-test 'beginend-dired-mode-goto-end filename))
+
+(defmacro with-beginend-dired-test-buffer (entries &rest body)
+  "Create a temporary directory with ENTRIES and execute BODY.
+
+ENTRIES is the file list.  The order is important.  ENTRIES are listed in that
```

Comment:
what about: "ENTRIES is a list of all filenames to display, in the same order, in the dired buffer."?